src/Model/Photo.php

```diff
@@ -173,4 +174,82 @@ public static function importProfilePhoto($photo, $uid, $cid, $quit_on_error = f
 
 		return array($photo, $thumb, $micro);
 	}
+
+	/**
+	 * @param string $exifCoord coordinate
+	 * @param string $hemi      hemi
+	 * @return float
+	 */
+	public static function getGps($exifCoord, $hemi)
+	{
+		$degrees = count($exifCoord) > 0 ? self::gps2Num($exifCoord[0]) : 0;
+		$minutes = count($exifCoord) > 1 ? self::gps2Num($exifCoord[1]) : 0;
+		$seconds = count($exifCoord) > 2 ? self::gps2Num($exifCoord[2]) : 0;
+	
+		$flip = ($hemi == 'W' || $hemi == 'S') ? -1 : 1;
+	
+		return floatval($flip * ($degrees + ($minutes / 60) + ($seconds / 3600)));
+	}
+
+	/**
+	 * @param string $coordPart coordPart
+	 * @return float
+	 */
+	private static function gps2Num($coordPart)
+	{
+		$parts = explode('/', $coordPart);
+	
+		if (count($parts) <= 0) {
+			return 0;
+		}
+	
+		if (count($parts) == 1) {
+			return $parts[0];
+		}
+	
+		return floatval($parts[0]) / floatval($parts[1]);
+	}
+
+	/**
+	 * @brief Fetch the photo albums that are available for a viewer
+	 *
+	 * The query in this function is cost intensive, so it is cached.
+	 *
+	 * @param int  $uid    User id of the photos
+	 * @param bool $update Update the cache
+	 *
+	 * @return array Returns array of the photo albums
+	 */
+	public static function photoAlbums($uid, $update = false)
```

Comment:
Please rename to `Photo::getAlbums`.

Please create a new function `Photo::clearAlbumCache()` that only unset the cache value using the same key as this function, and that should replace any `Photo::getAlbums` call with the `$update` parameter set to true where we don't use the return value.

Reply:
unset cache value = set value to null?

Reply:
Yup!